xtask/rules_check/src/lib.rs

```diff
@@ -0,0 +1,740 @@
+                            write!(content, ">")?;
+                        }
+                        LinkType::Inline | LinkType::Reference | LinkType::Shortcut => {
+                            write!(content, "]({dest_url}")?;
+                            if !title.is_empty() {
+                                write!(content, " \"{title}\"")?;
+                            }
+                            write!(content, ")")?;
+                        }
+                        _ => {
+                            panic!("unimplemented link type")
+                        }
+                    }
+                    start_link_tag = None;
+                } else {
+                    panic!("missing start link tag");
+                }
+            }
+
+            Event::SoftBreak => {
+                writeln!(content)?;
+            }
+
+            Event::HardBreak => {
+                writeln!(content, "<br />")?;
+            }
+
+            Event::Start(Tag::List(num)) => {
+                list_indentation += 1;
+                if let Some(num) = num {
+                    list_order = Some(num);
+                }
+                if list_indentation > 1 {
+                    writeln!(content)?;
+                }
+            }
+
+            Event::End(TagEnd::List(_)) => {
+                list_order = None;
+                list_indentation -= 1;
+                writeln!(content)?;
+            }
+            Event::Start(Tag::Item) => {
+                write!(content, "{}", "  ".repeat(list_indentation - 1))?;
+                if let Some(num) = list_order {
+                    write!(content, "{num}. ")?;
+                } else {
+                    write!(content, "- ")?;
+                }
+            }
+
+            Event::End(TagEnd::Item) => {
+                list_order = list_order.map(|item| item + 1);
+                writeln!(content)?;
+            }
+
+            Event::Start(Tag::Strong) => {
+                write!(content, "**")?;
+            }
+
+            Event::End(TagEnd::Strong) => {
+                write!(content, "**")?;
+            }
+
+            Event::Start(Tag::Emphasis) => {
+                write!(content, "_")?;
+            }
+
+            Event::End(TagEnd::Emphasis) => {
+                write!(content, "_")?;
+            }
+
+            Event::Start(Tag::Strikethrough) => {
+                write!(content, "~")?;
+            }
+
+            Event::End(TagEnd::Strikethrough) => {
+                write!(content, "~")?;
+            }
+
+            Event::Start(Tag::BlockQuote) => {
+                write!(content, ">")?;
+            }
+
+            Event::End(TagEnd::BlockQuote) => {
+                writeln!(content)?;
+            }
+
+            _ => {
+                // TODO: Implement remaining events as required
+                bail!("unimplemented event {event:?}")
+            }
+        }
+    }
+
+    Ok(())
+}
```

Comment:
I think we can remove most of the markdown content generation part in this function:

```suggestion
fn parse_documentation(
    group: &'static str,
    rule: &'static str,
    docs: &'static str,
    content: &mut Vec<u8>,
    has_fix_kind: bool,
) -> anyhow::Result<()> {
    let parser = Parser::new(docs);

    // Tracks the content of the current code block if it's using a
    // language supported for analysis
    let mut language = None;

    for event in parser {
        match event {
            // CodeBlock-specific handling
            Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(meta))) => {
                // Track the content of code blocks to pass them through the analyzer
                let test = CodeBlockTest::from_str(meta.as_ref())?;
                language = Some((test, String::new()));
            }
            Event::End(TagEnd::CodeBlock) => {
                if let Some((test, block)) = language.take() {
                    assert_lint(group, rule, &test, &block, has_fix_kind)?;
                }
            }
            Event::Text(text) => {
                if let Some((_, block)) = &mut language {
                    write!(block, "{text}")?;
                }
            }
            // We don't care other events
            _ => {}
        }
    }

    Ok(())
}
```

Reply:
Did you test it locally?

Reply:
> Did you test it locally?

No, but I can create a PR if this actually works.

Reply:
No, it's OK, I can do a follow-up